### PR TITLE
Fix instantiation stub maker for case where no register is used by function parameters

### PR DESCRIPTION
**Pull request by fadimounir**
Fix for #24701 

The issue here is that the stub maker logic always assumed there was at least one register used to pass parameters to the function after inserting the instantiation parameter. This is incorrect in the cases where more than one register is needed to pass a single argument, and there aren't enough registers available, in which case the parameter is passed on the stack (and some registers remain unused). 

Example:
- r0 used by this ptr
- r1 used by return buffer ptr
- r2 used by the inserted instantiation argument
- r3 unused
- stack0 + stack1 used to pass a `long` parameter

### CODE REVIEW

**Comment by fadimounir** [comment]
@jkotas PTAL

**Comment by jkotas** [comment]
Do we need a test?

File: src/vm/arm/stubs.cpp [hunk 1]
```diff
@@ -1824,33 +1824,42 @@ void StubLinkerCPU::ThumbEmitCallWithGenericInstantiationParameter(MethodDesc *p
...
+
+            // Finally we can swap the descriptors.
+            DWORD idxFirstRegDesc = 0;
+            while (cSwaps)
+            {
+                ArgDesc sTempDesc = rgArgDescs[idxLastRegDesc];
+                rgArgDescs[idxLastRegDesc] = rgArgDescs[idxFirstRegDesc];
+                rgArgDescs[idxFirstRegDesc] = sTempDesc;
+
+                _ASSERTE(idxFirstRegDesc < ((DWORD)idxLastRegDesc));
```
Comment (jkotas) [inline src/vm/arm/stubs.cpp]: Change `idxLastRegDesc1` to `int` as well to avoid this cast?

**Comment by jkotas** [comment]
LGTM otherwise

**Comment by fadimounir** [comment]
> Do we need a test?

Isn't this already covered by these PMI tests? I can always add a new test if that's not the case. These PMI tests seemed to reflect over all APIs, probably for each runtime assembly, so it should be pretty good coverage

**Comment by jkotas** [comment]
@dotnet/jit-contrib Is it ok to depend on PMI tests for test coverage for regressions; or do we need usually add a specific regression tests?

**Comment by fadimounir** [comment]
I'm going to add one anyways, just in case

**Comment by CarolEidt** [comment]
I think that adding a test is the right thing to do - AFAIK we don't (yet?) have pmi testing in the pipelines. And, even if we did, it would be prudent to have something that actually tested that it was doing the right thing :-)

**Comment by CarolEidt** [comment]
The fix looks good to me, but should presumably be reviewed by someone who's actually familiar with this code.

**Comment by AndyAyersMS** [comment]
PMI doesn't usually run the code that gets generated, so it would be a good to add a test.

**Pull request merged**